packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1717,9 +1718,8 @@ export class ThinEngine {
 
         this._activeRenderLoops.push(renderFunction);
 
-        if (!this._renderingQueueLaunched) {
-            this._renderingQueueLaunched = true;
-            this._boundRenderFunction = () => this._renderLoop();
+        // On the first added function, start the render loop.
+        if (this._activeRenderLoops.length === 1 && this._frameHandler === -1) {

```

Comment:
I think you don't need the `this._activeRenderLoops.length === 1` part? If `this._frameHandler == -1`, it means no RAF is in flight.

Reply:
I was concerned by 
![image](https://github.com/BabylonJS/Babylon.js/assets/16655165/538c2e6c-901a-4e12-949e-370fac7c12a9)

For back compat but I suppose I should make the list readonly so we can get read of the check ?